src/internal/operations/backup.go

```diff
@@ -655,88 +763,68 @@ func mergeDetails(
 	writeStats.TotalNonMetaUploadedBytes = detailsModel.SumNonMetaFileSizes()
 
 	// Don't bother loading any of the base details if there's nothing we need to merge.
-	if dataFromBackup == nil || dataFromBackup.ItemsToMerge() == 0 {
+	if bases == nil || dataFromBackup == nil || dataFromBackup.ItemsToMerge() == 0 {
 		return nil
 	}
 
-	var addedEntries int
-
-	for _, baseBackup := range backups {
-		var (
-			mctx                 = clues.Add(ctx, "base_backup_id", baseBackup.ID)
-			manifestAddedEntries int
-		)
+	var (
+		addedEntries int
+		// alreadySeenEntries tracks items that we've already merged so we don't
+		// accidentally merge them again. This could happen if, for example, there's
+		// an assist backup and a merge backup that both have the same version of an
+		// item at the same path.
+		alreadySeenEntries = map[string]struct{}{}
+	)
 
-		baseDeets, err := getDetailsFromBackup(
-			mctx,
-			baseBackup.Backup,
+	// Just to be on the safe side merge details from assist bases first. It
```

Comment:
You say that twice in the same paragraph. (:
```suggestion
	// Merge details from assist bases first. It
```